tutorials/setting-up-ruby-discourse.md

```diff
@@ -0,0 +1,168 @@
+
+## Configure Cloud SQL Postgres instance
+
+We're going to create a database named `discourse` on the Cloud SQL Postgres instance. This is going to be the production
+database your Discourse app will use.
+
+1. In the Cloud Platform Console, go to the [Cloud SQL Postgres](https://console.cloud.google.com/sql/instances) instance you have created.
+
+1. Go to the **Databases** tab. Create a new database with name `discourse`.
+
+1. Go to **Access Control** tab, and then **Users** sub-tab. Create a new user with username `discourse` and password `discourse`.
+
+1. Go back to the **Overview** tab, and click on the **Connect using Cloud Shell** button.
+
+1. In the Cloud Shell terminal that appears, run:
+
+  ```sh
+  gcloud beta sql connect <The Cloud SQL instance name> --user=discourse
+  ```
+
+1. When prompted, type in the password `discourse`.
+
+1. After you connect to the instance, run:
+
+  ```sh
+  discourse=> GRANT ALL PRIVILEGES ON DATABASE discourse TO discourse;
+  ```
+
+1. Close the Cloud Shell when done.
+
+1. On the same **Overview** page, note the **Instance connection name** for later use.
+
+## Configure Discourse Rails application
+
+To download and install the Discourse Rails application locally, 
+follow the official [Discourse Advanced Developer Install Guide](https://github.com/discourse/discourse/blob/master/docs/DEVELOPER-ADVANCED.md). You can
+also set up the local Postgres database and Redis server to test the app on your local machine in a development environment. 
+We'll configure the Discourse app to run on the App Engine flexible environment with the production environment.
+
+1. Under the `config/` directory in the Discourse app, create a file with name `discourse.conf`:
+
+  ```
+  # password used to access the db
+  db_password = discourse
+
+  # socket name for database connection
+  db_host = /cloudsql/<Cloud SQL instance connection name>
+
+  # Redis host address
+  redis_host = <Redis instance internal IPv4 address>
+
+  # enable serve_static_assets for dockerized app
+  serve_static_assets = true
+  ```
+
+  This file sets configuration parameters in the production environment for Discourse.
+
+1. Open the `config/puma.rb` file, then change `APP_ROOT = '/home/discourse/discourse':
+
+  ```ruby
+  APP_ROOT = Rails.root
+  ```
+1. In the same `config/puma.rb` file, delete the line that says `daemonize true`.
+   This is because we shouldn't run a daemonized process in Docker container.
+
+1. Save and close the file.
+
+
+## Deploy Discourse app to the App Engine flexible environment
+
+
+1. Under the application root directory of the Discourse app, create a file named `app.yaml`:
+  ```
+    runtime: ruby
+    env: flex
+    entrypoint: bundle exec rails s -p 8080
+    beta_settings:
+      cloud_sql_instances: <Cloud SQL Instance connection name>
+  ```
+
+1. In the same application root directory, run this gcloud SDK command to deploy:
+  ```sh
+    VERSION=$(date +%Y%m%dt%H%M%S); \
```

Comment:
Makes sense. It might be a good idea to add that rationale below the command to make it clear to others as well.